server/channels/app/channel_category.go

```diff
@@ -25,19 +25,23 @@ func (a *App) createInitialSidebarCategories(userID string, opts *store.SidebarC
 
 func (a *App) GetSidebarCategoriesForTeamForUser(c request.CTX, userID, teamID string) (*model.OrderedSidebarCategories, *model.AppError) {
 	var appErr *model.AppError
-	categories, err := a.Srv().Store().Channel().GetSidebarCategoriesForTeamForUser(userID, teamID)
-	if err == nil && len(categories.Categories) == 0 {
-		// A user must always have categories, so migration must not have happened yet, and we should run it ourselves
-		categories, appErr = a.createInitialSidebarCategories(userID, &store.SidebarCategorySearchOpts{
-			TeamID:      teamID,
-			ExcludeTeam: false,
-		})
-		if appErr != nil {
-			return nil, appErr
-		}
+	appsCategoryEnabled := a.Config().FeatureFlags.AppsSidebarCategory
+	options := &store.SidebarCategorySearchOpts{
+		TeamID:              teamID,
+		ExcludeTeam:         false,
+		AppsCategoryEnabled: appsCategoryEnabled,
 	}
-
-	if err != nil {
+	categories, err := a.Srv().Store().Channel().GetSidebarCategoriesForTeamForUser(userID, teamID, options)
+	if err == nil {
```

Comment:
Agreed, the nested if could cause confusion. Updated to just add the condition to the existing conditional.

Reply:
The change you've applied looks good, though the change I was proposing is to move this block after the `err != nil` block, which always returns. Something like this:

```go
if err != nil {
    // ...
    return err
}

if len(categories.Categories) == 0 || (appsCategoryEnabled && checkMissingSystemSidebarCategories(categories)) {
    // ...
}
```

0/5 on this change, just a preference of mine to avoid duplicate error checking, since the rest of the function already assumes `err == nil`